lib/less/rails/railtie.rb

```diff
@@ -14,11 +14,16 @@ class Railtie < ::Rails::Railtie
         require 'less-rails'
         Sprockets::Engines #force autoloading
 
-        if Sprockets.respond_to?(:register_engine)
-          args = ['.less', LessTemplate]
-          args << { mime_type: 'text/less', silence_deprecation: true } if Sprockets::VERSION.start_with?("3")
-          env.register_engine(*args)
-        end
+        config.assets.configure do |env|
+          if env.respond_to?(:register_engine)
+            args = ['.less', LessTemplate]
+            args << { mime_type: 'text/less', silence_deprecation: true } if env::VERSION.start_with?("3")
+            env.register_engine(*args)
+          elsif env.respond_to?(:register_transformer)
```

Comment:
Very strange to check `register_transformer` method and don't use it. Will be better to find version when this deprecation warning was added and use comparison with this version instead.